flocker_bb/builders/flocker.py

```diff
@@ -573,6 +573,21 @@ def makeHomebrewRecipeTestFactory():
 
 def getBuilders(slavenames):
     builders = [
+        BuilderConfig(
+            name='flocker/functional/storage-driver',
+            builddir='flocker',
+            slavenames=slavenames[
+                'centos-7',
+                'ubuntu-14.04',
+                'flocker/functional/rackspace/centos-7/storage-driver'
+            ],
```

Comment:
This part doesn't quite make sense.  `slavenames` is just a mapping from semi-generic slave type identifiers to lists of actual names of actual slaves which fit that "type".  Indexing into it with a three-tuple like this is guaranteed to fail.

Instead, you want to make a builder for each of these platforms.  Or rather, for each of these platforms:

  * rackspace, centos-7
  * rackspace, ubuntu 14.04 (but we don't have a slave image for this yet so you can't enable it yet)
  * aws, centos-7
  * aws, ubuntu 14.04 (maybe we don't have this image?  I don't know.  Maybe we do)

And give each builder some appropriate slaves.  For the rackspace builders, give it any rackspace slaves.  For the AWS builders, give it any EC2 slaves.